worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -362,7 +362,7 @@ private long getRollOverThreshold() {
                 }
 
                 // Add the thread to process the patient and start the thread
-                futureHandles.add(patientClaimsProcessor.process(patient, helper, contract.getAttestedOn()));
+                futureHandles.add(patientClaimsProcessor.process(contractNumber, patient, helper, contract.getAttestedOn()));
```

Comment:
May be use a `ThreadLocal` to pass contract number to RoundRobinBlockingQueue without having to modify method signatures or to bring in custom Executor implementations.